qualification/report/katdoc.sty

```diff
@@ -1,64 +1,84 @@
-% File: katdoc.sty
-% Subject: Stylesheet for KAT document
-% Author: M. Welz and A. Pinska; based on template by R.T. Lord and J.C. Guzman
 \ProvidesPackage{katdoc}
-% \setcounter{errorcontextlines}{999}
-
-\newcommand{\stylesheetVersion}{1.1.2-dirty}
+\newcommand{\stylesheetVersion}{1.1.3-DSP}
 
+\usepackage[utf8]{inputenc}
 \usepackage{graphicx}
 \usepackage{lastpage}
 \usepackage{mathptmx}
 \usepackage[T1]{fontenc}
 \usepackage{geometry}
 \usepackage{array}
 \usepackage{babel}
+\usepackage{csquotes}  % Latex recommends this when babel is used
 \usepackage{fancyhdr}
 \usepackage{tabularx}
 \usepackage{longtable}
 \usepackage{multirow}
 \usepackage{ifthen}
 \usepackage{float}
 \usepackage{enumitem}
+\usepackage{mathtools}
+\usepackage{tablefootnote}
+\usepackage{color}
+\usepackage{colortbl}
 
 % For plotting to work nicely.
-\usepackage[utf8]{inputenc}
+\usepackage{pgf}
 \usepackage{pgfplots}
 \DeclareUnicodeCharacter{2212}{−}
 \usepgfplotslibrary{groupplots,dateplot}
 \usetikzlibrary{patterns,shapes.arrows}
 \pgfplotsset{compat=newest}
 
-\geometry{verbose,a4paper,tmargin=32mm,bmargin=27mm,lmargin=21mm,rmargin=19mm,headheight=10mm,headsep=12mm,footskip=12mm}
-\pagestyle{empty}
+\geometry{nomarginpar,verbose,a4paper,tmargin=32mm,bmargin=27mm,lmargin=21mm,rmargin=19mm,headheight=10mm,headsep=12mm,footskip=12mm}
 \setlength{\parskip}{\medskipamount}
 \setlength{\parindent}{0pt}
-\pagenumbering{arabic}
 \setlength{\unitlength}{1mm}
 \setcounter{secnumdepth}{5}
 \setcounter{tocdepth}{5}
 
-% Template revision
-\newcommand{\templateVersion}{???}
-
 % Front Cover
-
-\newcommand{\addcontributor}[5]{
+\newcommand{\addcontributor}[5]{%
   #1 & #2 & #3 & #4 & #5 & \\
   \hline
 }
 
-\newcommand{\addchange}[4]{
+\newcommand{\addchange}[4]{%
   #1 & #2 & #3 & #4 \\
   \hline
 }
 
-\newcommand{\addprogram}[4]{
+\newcommand{\addprogram}[4]{%
   #1 & #2 & #3 \\
   \hline
 }
 
-\renewcommand{\maketitle}{
+% Abbreviations
+
+\newcommand{\abbrev}[2]{%
+  #1 & #2\\
+  \hline
+}
+
+\newcommand{\listofabbreviations}{%
+  \section*{Glossary}
+
+  \begin{tabular}{|p{0.1\textwidth}|p{0.84\textwidth}|}
+    \hline
+    \abbreviations%
+  \end{tabular}
+}
+
+% Extract version information from pdftex
+\def\extractpdftexversion#1-#2-#3 #4\@nil{#3}%
+\def\myversion{\expandafter\extractpdftexversion\pdftexbanner\@nil}%
```

Comment:
Is it worth trying to protect this so that it produces something sane when compiled with a different engine?

Reply:
I don't see myself ever bothering to use a different engine.

It's probably better practice to do so, but I can forsee myself spending some hours figuring out how to do that, which is not really an exercise that I'm interested in right now.

Reply:
Just tested with lualatex and it reports that `\pdftexbanner` is an undefined control sequence, which I think is an acceptable outcome (it doesn't produce a document with incorrect information, and the error is a fairly good hint). You can detect whether the engine is pdftex with `\ifpdftex`, but I found it wasn't all that helpful because while you can emit an error message, latexmk keeps rolling on and it gets lost in amongst all the other errors.

Reply:
The Internet showed me a simpler solution: add `\RequirePDFTeX` (e.g. after importing all the other packages - it's provided by the `iftex` package so for completeness you could import that). If using the wrong engine, it prints the following then aborts:
```
 ********************************************
 * pdfTeX is required to compile this document.
 * Sorry!
 ********************************************
```